### ISSUE TITLE
Are nested color objects in theme supported?

### ISSUE THREAD

**Issue by YassienW** [styled-components/xstyled#413]
I wasn't sure if i should file a bug report or not, so i'm asking first. I organize my colors as follows:

```javascript
const theme: AppTheme = {
  colors: {
    blue: {
      1: '#83879c',
      2: '#B6B9CB'
      //3: etc..
    }
  }
}
```
Considering a simple button component that uses the xstyled color prop:
```javascript
<Button color="blue.1"/>
```

I get this typescript error: 

![image](https://github.com/styled-components/xstyled/assets/11986226/950cd1c2-359c-4e55-b53a-62b28a18c8c9)

Should i change the way i organize my colors because this is expected? Or should it not throw a type error? The color itself works as it should, but the type is causing the problem
